runtime/command-metrics/src/main/java/io/aklivity/zilla/runtime/command/metrics/internal/airline/ZillaMetricsCommand.java

```diff
@@ -128,67 +78,68 @@ private Path engineDirectory()
                 rethrowUnchecked(ex);
             }
         }
-        EngineConfiguration config = new EngineConfiguration(props);
-        return config.directory();
-    }
-
-    private List<MetricsLayout> countersLayouts() throws IOException
-    {
-        Stream<Path> files = Files.walk(metricsPath, 1);
-        return files.filter(this::isCountersFile).map(this::newCountersLayout).collect(toList());
-    }
-
-    private List<MetricsLayout> gaugesLayouts() throws IOException
-    {
-        Stream<Path> files = Files.walk(metricsPath, 1);
-        return files.filter(this::isGaugesFile).map(this::newGaugesLayout).collect(toList());
-    }
-
-    private List<MetricsLayout> histogramsLayouts() throws IOException
-    {
-        Stream<Path> files = Files.walk(metricsPath, 1);
-        return files.filter(this::isHistogramsFile).map(this::newHistogramsLayout).collect(toList());
-    }
+        final Configuration config = new EngineConfiguration(props);
+        final ErrorHandler onError = Throwable::printStackTrace;
+        final String binding = args != null && args.size() >= 1 ? args.get(0) : null;
 
-    private boolean isCountersFile(
-        Path path)
-    {
-        return path.getNameCount() - metricsPath.getNameCount() == 1 &&
-                COUNTERS_PATTERN.matcher(path.getName(path.getNameCount() - 1).toString()).matches() &&
-                Files.isRegularFile(path);
-    }
-
-    private boolean isGaugesFile(
-        Path path)
-    {
-        return path.getNameCount() - metricsPath.getNameCount() == 1 &&
-                GAUGES_PATTERN.matcher(path.getName(path.getNameCount() - 1).toString()).matches() &&
-                Files.isRegularFile(path);
-    }
-
-    private boolean isHistogramsFile(
-        Path path)
-    {
-        return path.getNameCount() - metricsPath.getNameCount() == 1 &&
-                HISTOGRAMS_PATTERN.matcher(path.getName(path.getNameCount() - 1).toString()).matches() &&
-                Files.isRegularFile(path);
+        do
+        {
+            try (Engine engine = Engine.builder()
+                .config(config)
+                .errorHandler(onError)
+                .readonly()
+                .build())
+            {
+                engine.start();
+                MetricsReader metrics = new MetricsReader(engine, engine::supplyLocalName);
+                LongPredicate filterPredicate = filterPredicate(namespace, binding, engine::supplyLabelId);
+                List<MetricRecord> records = filter(metrics.records(), filterPredicate);
+                MetricsPrinter printer = new MetricsPrinter(records);
+                printer.print(System.out);
+            }
+            catch (Throwable ex)
+            {
+                System.out.println("error");
+                rethrowUnchecked(ex);
+            }
+            sleep(interval);
+        } while (interval != 0);
```

Comment:
Well, I see a tradeoff here. All that you wrote are on the pro side of the suggested change. On the con side there is the fact that zilla live reloads the changes in the `zilla.yaml` config file that includes the list of metrics.

With the change you suggest, if someone adds a new metric to the config while the zilla metrics command is running in continuous mode, it won't pick that up (until stopped and started again). Also, I believe it would display incorrect information if someone removed a metric from the config while it is running.

Since the metrics command has no awareness of the config of the running instance, and we do not store metrics metadata in the data files, we have no other way to detect if the list of metrics has changed but reopening the engine.

One way to mitigate this issue is to document the requirement that it's necessary to stop and restart the continuously running metrics command if the list of metrics has been changed by the user in the config of the running instance.

Having said all this, plus your note about the `watch` command, I guess it's also a legit question if we want to keep this feature around at all.

Pls LMK.

Reply:
Agreed, I think our choices are
  - remove the interval concept and use `watch` instead, OR
  - move the loop inside the `try-with-resources` and treat the change detection as a bug to be resolved

I think it is good enough to go with the first option for now, and just remove the looping.

Reply:
OK, removed the loop and the interval option.